File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
import sys

n, m = map(int, sys.stdin.readline().split())
arr = []
cnt = []

for _ in range(n):
    arr.append(sys.stdin.readline().rstrip())

for i in range(n-7):
    for j in range(m-7):
        b_cnt = 0
        w_cnt = 0

        for k in range(i, i+8):
            for l in range(j, j+8):
                if (k+l) % 2 == 0:
                    if arr[k][l] != "B":
                        b_cnt += 1
                    if arr[k][l] != "W":
                        w_cnt += 1
                else:
                    if arr[k][l] != "W":
                        b_cnt += 1
                    if arr[k][l] != "B":
                        w_cnt += 1

        cnt.append(b_cnt)
        cnt.append(w_cnt)
        
sys.stdout.write(f"{min(cnt)}\n")

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    
    int n, m;
    int min_cnt = 65;
    cin >> n >> m;
    vector<string> arr(n);

    for (string& colors : arr)
        cin >> colors;

    for (int i = 0; i < n-7; i++) {
        for (int j = 0; j < m-7; j++) {
            int b_cnt = 0, w_cnt = 0;

            for (int k = i; k < i+8; k++) {
                for (int l = j; l < j+8; l++) {
                    if ((k+l) % 2 == 0) {
                        if (arr[k][l] != 'B')
                            b_cnt++;
                        if (arr[k][l] != 'W')
                            w_cnt++;
                    }
                    else {
                        if (arr[k][l] != 'W')
                            b_cnt++;
                        if (arr[k][l] != 'B')
                            w_cnt++;
                    }
                }
            }

            if (b_cnt <= w_cnt && b_cnt < min_cnt)
                min_cnt = b_cnt;
            else if (w_cnt < b_cnt && w_cnt < min_cnt)
                min_cnt = w_cnt;
        }
    }

    cout << min_cnt << "\n";
    
    return 0;
}